lib/auth.js

```diff
@@ -1,10 +1,62 @@
 'use strict';
 
+const assert = require('assert');
 const fs = require('fs');
 const os = require('os');
 const path = require('path');
+const util = require('util');
+const readline = require('readline');
+const readFile = util.promisify(fs.readFile);
+const writeFile = util.promisify(fs.writeFile);
 const authFile = path.join(os.homedir(), '.ncurc');
-// TODO: try-catch, validate properties
-const { username, token } = JSON.parse(fs.readFileSync(authFile, 'utf8'));
-const auth = Buffer.from(`${username}:${token}`).toString('base64');
 module.exports = auth;
+
+function check(username, token) {
+  assert(typeof username === 'string' && /^[a-zA-Z0-9]*/.test(username));
+  assert(typeof token === 'string' && /^[0-9a-f]*/.test(token));
+}
+
+async function auth() {
+  let username, token;
+  // Try reading from ~/.ncurc
+  try {
+    ({ username, token } = JSON.parse(await readFile(authFile, 'utf8')));
+  } catch (e) {
+    process.stdout.write('Reading configuration for node-core-utils failed:\n' +
+                         e.message + '\n');
+  }
+
+  // If that worked, yay
+  if (username && token) {
+    check(username, token);
+    return Buffer.from(`${username}:${token}`).toString('base64');
+  }
+
+  // Ask the user for input and write to ~/.ncurc, then try again
+  process.stdout.write('Please enter your Github user information:\n' +
+      '[Github tokens can be created as described in ' +
+      'https://help.github.com/articles/' +
+      'creating-a-personal-access-token-for-the-command-line/]\n');
+  username = await prompt('Github user name');
+  token = await prompt('Github token');
+  check(username, token);
+  const json = JSON.stringify({ username, token }, null, '  ');
+  await writeFile(authFile, json);
```

Comment:
should set some reasonable default permissions on this file perhaps? owner read+write only perhaps?

Reply:
yes. :) done!